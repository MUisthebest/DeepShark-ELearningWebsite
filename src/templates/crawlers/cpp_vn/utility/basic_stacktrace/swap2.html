 <h1 class="firstHeading" id="firstHeading">chuyên về<small>Thuật toán</small>
</h1> <table class="t-dcl-begin"> <tr class="t-dsc-header"> <th>C Thư viện<code><a href="../../header/stacktrace" title="cpp/header/stacktrace">&lt;stacktrace&gt;</a></code> </th> <th> </th> <th> </th> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">Mẫu &lt;CLASE ALTSATOR&gt;</pre>
</td> <td class="t-dcl-nopad"> </td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> </table> <p>Mẫu &lt;class T, std :: size_t n&gt;<code><a href="../../algorithm/swap" title="cpp/algorithm/swap">std::swap</a></code>Void hoán đổi (std :: mảng &lt;t, n&gt; &amp; lhs,<code>std::basic_stacktrace</code>           std :: mảng &lt;t, n&gt; &amp; rhs)<code>lhs</code>Và<code>rhs</code>là các yếu tố của phạm vi.<code>lhs.swap(rhs);</code>.</p>
<h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>E lvalues</td> <td>-</td> <td>Void hoán đổi (std :: basic_stacktrace &lt;allocator&gt; &amp; lhs, std :: basic_stacktrace &lt;folocator&gt; &amp; rhs)</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p>Gán cho đã cho</p>
<h3 id="Complexity">Không hợp lệ bất kỳ trình lặp lại nào (bao gồm bất kỳ trình lặp cuối nào) hoặc sửa đổi bất kỳ yếu tố nào của các phạm vi liên quan, hành vi không được xác định.</h3> <p>trình rút vào các yếu tố để hoán đổi</p>
<h3 id="Example">Ví dụ</h3> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="swap" title="cpp/utility/basic stacktrace/swap"> <span class="t-lines"><span>Được xác định trong tiêu đề</span></span></a></div> </td> <td>Trả về số lượng phần tử tối đa có thể<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/basic_stacktrace/swap2">https://en.cppreference.com/w/cpp/utility/basic_stacktrace/swap2</a>
</p>
</div>
