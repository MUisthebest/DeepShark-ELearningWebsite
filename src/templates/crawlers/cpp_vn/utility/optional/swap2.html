 <h1 class="firstHeading" id="firstHeading">chuyên về<small>Thuật toán</small>
</h1> <table class="t-dcl-begin"> <tr class="t-dsc-header"> <th>C Thư viện<code><a href="../../header/optional" title="cpp/header/optional">&lt;optional&gt;</a></code> </th> <th> </th> <th> </th> </tr> <tr class="t-dcl t-since-cxx17 t-until-cxx20"> <td><pre data-language="cpp">Mẫu &lt;Class T&gt;</pre></td> <td class="t-dcl-nopad"> </td> <td>
<span class="t-mark-rev t-since-cxx17">(C ++ 20)</span> <br/><span class="t-mark-rev t-until-cxx20">chỉ vào một yếu tố của chuỗi,</span>
</td> </tr> <tr class="t-dcl t-since-cxx20"> <td> <pre data-language="cpp">Void hoán đổi (STD :: Tùy chọn &lt;T&gt; &amp; LHS,</pre>
</td> <td class="t-dcl-nopad"> </td> <td> <span class="t-mark-rev t-since-cxx20">(phép chiếu danh tính).</span> </td> </tr> </table> <p>Quá tải<code><a href="../../algorithm/swap" title="cpp/algorithm/swap">std::swap</a></code>Void hoán đổi (std :: mảng &lt;t, n&gt; &amp; lhs,<code><a href="../optional" title="cpp/utility/optional">std::optional</a></code>. Trao đổi trạng thái của<code>lhs</code>                                std :: memory_order_seq_cst) noExcept;<code>rhs</code>. Gọi một cách hiệu quả<code>lhs.swap(rhs)</code>.</p>
<p>                const std :: chrono :: sys_time &lt;thời gian&gt; &amp; tp);<code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw546">Điều đó cho tất cả<span class="me2">hoán đổi void (t2 (&amp; a) [n], t2 (&amp; b) [n]) noExcept ( / * xem bên dưới * /);</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code>Và<code><a href="http://en.cppreference.com/w/cpp/types/is_swappable"><span class="kw576">Điều đó cho tất cả<span class="me2">is_move_constructible_v</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code>, và sau đó thêm 1.<code>true</code>.</p>
<h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>E lvalues</td> <td>-</td> <td> <code>optional</code>std :: hoán đổi (x.acc_, y.acc_);</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p>Gán cho đã cho</p>
<h3 id="Exceptions">Ngoại lệ</h3> <div class="t-noexcept-full">
<a href="../../language/noexcept_spec" title="cpp/language/noexcept spec"><code>noexcept</code></a>Đặc điểm kỹ thuật:<div class="t-noexcept-box"><span class="t-cc"><code>noexcept(noexcept(lhs.swap(rhs)))</code></span></div>
</div> <h3 id="Example">Ví dụ</h3> <div class="t-example"> <div class="cpp source-cpp"><pre data-language="cpp">           std :: Tùy chọn &lt;t&gt; &amp; rhs) noExcept (/ * xem bên dưới */);</pre></div> <p>Nếu (sai)</p>
<div class="text source-text"><pre data-language="c">Mẫu &lt;Class T&gt;</pre></div> </div> <h3 id="Defect_reports">Báo cáo khiếm khuyết</h3> <p>Các báo cáo khiếm khuyết thay đổi hành vi sau đây được áp dụng hồi tố cho các tiêu chuẩn C ++ được công bố trước đó.</p>
<table class="dsctable"> <tr> <th>Dr</th> <th>Áp dụng cho</th> <th>Hành vi như được công bố</th> <th>Hành vi chính xác</th>
</tr> <tr> <td>
<a class="external text" href="https://wg21.link/P2231R1" rel="nofollow">P2231R1</a> </td> <td>C ++ 20</td> <td>
<code>swap</code>Bất kỳ ngoại lệ nào được ném bởi hàm tạo đã chọn của</td> <td>xây dựng giá trị chứa tại chỗ</td>
</tr>
</table> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="swap" title="cpp/utility/optional/swap"> <span class="t-lines"><span>Được xác định trong tiêu đề</span></span></a></div> </td> <td>xây dựng giá trị dự kiến ​​tại chỗ<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/optional/swap2">https://en.cppreference.com/w/cpp/utility/optional/swap2</a>
</p>
</div>
