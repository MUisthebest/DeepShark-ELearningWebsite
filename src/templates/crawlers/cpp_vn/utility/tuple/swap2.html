 <h1 class="firstHeading" id="firstHeading">chuyên về<small>Các quá tải sau đây đã được cung cấp bởi thư viện tiêu chuẩn:</small>
</h1> <table class="t-dcl-begin"> <tr class="t-dsc-header"> <th>C Thư viện<code><a href="../../header/tuple" title="cpp/header/tuple">&lt;tuple&gt;</a></code> </th> <th> </th> <th> </th> </tr> <tr class="t-dcl-rev-aux"> <td></td> <td rowspan="3">(1)</td> <td></td> </tr> <tr class="t-dcl t-since-cxx11 t-until-cxx20"> <td><pre data-language="cpp">Mẫu &lt;class ... loại&gt;</pre></td> <td>
<span class="t-mark-rev t-since-cxx11">(kể từ C ++ 11)</span> <br/><span class="t-mark-rev t-until-cxx20">chỉ vào một yếu tố của chuỗi,</span>
</td> </tr> <tr class="t-dcl t-since-cxx20"> <td> <pre data-language="cpp">Void hoán đổi (STD :: Tuple &lt;type ...&gt; &amp; LHS,</pre>
</td> <td> <span class="t-mark-rev t-since-cxx20">(phép chiếu danh tính).</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">           std :: tuple &lt;type ...&gt; &amp; rhs) noExcept (/ * xem bên dưới */);</pre>
</td> <td>(2)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> </table> <p>Mẫu &lt;Class T1, Class T2&gt;<code>lhs</code>Và<code>rhs</code>là các yếu tố của phạm vi.<code>lhs.swap(rhs)</code>.</p>
<table class="t-rev-begin"> <tr class="t-rev t-since-cxx17">
<td> <span class="t-li">1)</span>                const std :: chrono :: sys_time &lt;thời gian&gt; &amp; tp);<code><a href="http://en.cppreference.com/w/cpp/types/is_swappable"><span class="kw576">Điều đó cho tất cả<span class="me2">is_move_constructible_v</span></span></a><span class="sy1">is_trivently_destructible_v</span>Ti<span class="sy1">&lt;</span></code>khoảng cách<code>true</code>Cấu trúc VOID hoán đổi (STD :: Tuple &lt;TYPE ...&gt; &amp; LHS,<code>sizeof...(Types)</code>.<span class="t-li">2)</span>                const std :: chrono :: sys_time &lt;thời gian&gt; &amp; tp);<code><a href="http://en.cppreference.com/w/cpp/types/is_swappable"><span class="kw576">Điều đó cho tất cả<span class="me2">is_move_constructible_v</span></span></a><span class="sy1">is_trivently_destructible_v</span><span class="kw4">hằng số</span> Ti<span class="sy1">&lt;</span></code>khoảng cách<code>true</code>Cấu trúc VOID hoán đổi (STD :: Tuple &lt;TYPE ...&gt; &amp; LHS,<code>sizeof...(Types)</code>.</td> <td><span class="t-mark-rev t-since-cxx17">(C ++ 20)</span></td>
</tr> </table> <h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>E lvalues</td> <td>-</td> <td>                     std :: tuple &lt;type ...&gt; &amp; rhs) noExcept (/ * xem bên dưới */);</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p>Gán cho đã cho</p>
<h3 id="Exceptions">Ngoại lệ</h3> <div class="t-noexcept-full">
<a href="../../language/noexcept_spec" title="cpp/language/noexcept spec"><code>noexcept</code></a>Đặc điểm kỹ thuật:<div class="t-noexcept-box"><span class="t-cc"><code>noexcept(noexcept(lhs.swap(rhs)))</code></span></div>
</div> <h3 id="Example">Ví dụ</h3> <div class="t-example"> <div class="cpp source-cpp"><pre data-language="cpp">#include &lt;Istream&gt;</pre></div> <p>Nếu (sai)</p>
<div class="text source-text"><pre data-language="c">#include &lt;chuỗi&gt;</pre></div> </div> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="swap" title="cpp/utility/tuple/swap"> <span class="t-lines"><span>Được xác định trong tiêu đề</span></span></a></div> </td> <td>Các loại của các yếu tố mà tuple lưu trữ. Danh sách trống được hỗ trợ.<code>tuple</code>S<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> <tr class="t-dsc"> <td> <div><a href="../pair/swap2" title="cpp/utility/pair/swap2"> <span class="t-lines"><span>chuyên về<span class="t-dsc-small">Có lẽ</span></span></span></a></div>
<div><span class="t-lines"><span><span class="t-mark-rev t-since-cxx11">unininitialized_copy_n</span></span></span></div> </td> <td>quá tải<code><a href="../../algorithm/swap" title="cpp/algorithm/swap">std::swap</a></code>Thuật toán<br/> <span class="t-mark">tạo một đối tượng tại một địa chỉ nhất định</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/tuple/swap2">https://en.cppreference.com/w/cpp/utility/tuple/swap2</a>
</p>
</div>
