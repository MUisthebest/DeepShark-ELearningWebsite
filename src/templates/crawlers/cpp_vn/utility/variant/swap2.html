 <h1 class="firstHeading" id="firstHeading">chuyên về<small>Thuật toán</small>
</h1> <table class="t-dcl-begin"> <tr class="t-dsc-header"> <th>C Thư viện<code><a href="../../header/variant" title="cpp/header/variant">&lt;variant&gt;</a></code> </th> <th> </th> <th> </th> </tr> <tr class="t-dcl t-since-cxx17 t-until-cxx20"> <td><pre data-language="cpp">Mẫu &lt;class ... loại&gt;</pre></td> <td class="t-dcl-nopad"> </td> <td>
<span class="t-mark-rev t-since-cxx17">(C ++ 20)</span> <br/><span class="t-mark-rev t-until-cxx20">chỉ vào một yếu tố của chuỗi,</span>
</td> </tr> <tr class="t-dcl t-since-cxx20"> <td> <pre data-language="cpp">Void hoán đổi (std :: biến thể &lt;types ...&gt; &amp; lhs,</pre>
</td> <td class="t-dcl-nopad"> </td> <td> <span class="t-mark-rev t-since-cxx20">(phép chiếu danh tính).</span> </td> </tr> </table> <p>Quá tải<code><a href="../../algorithm/swap" title="cpp/algorithm/swap">std::swap</a></code>Void hoán đổi (std :: mảng &lt;t, n&gt; &amp; lhs,<code><a href="../variant" title="cpp/utility/variant">std::variant</a></code>. Gọi một cách hiệu quả<code>lhs.swap(rhs)</code>.</p>
<p>                const std :: chrono :: sys_time &lt;thời gian&gt; &amp; tp);<code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw546">Điều đó cho tất cả<span class="me2">hoán đổi void (t2 (&amp; a) [n], t2 (&amp; b) [n]) noExcept ( / * xem bên dưới * /);</span></span></a><span class="sy1">is_trivently_destructible_v</span>T_i<span class="sy1">&lt;</span></code>Và<code><a href="http://en.cppreference.com/w/cpp/types/is_swappable"><span class="kw576">Điều đó cho tất cả<span class="me2">is_move_constructible_v</span></span></a><span class="sy1">is_trivently_destructible_v</span>T_i<span class="sy1">&lt;</span></code>, và sau đó thêm 1.<code>true</code>cho tất cả<code>T_i</code>TRONG<code>Types...</code></p>
<h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>E lvalues</td> <td>-</td> <td> <code>variant</code>           std :: Variant &lt;type ...&gt; &amp; rhs) NoExcept (/ * xem bên dưới */);</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p>Gán cho đã cho</p>
<h3 id="Exceptions">Ngoại lệ</h3> <div class="t-noexcept-full">
<a href="../../language/noexcept_spec" title="cpp/language/noexcept spec"><code>noexcept</code></a>Đặc điểm kỹ thuật:<div class="t-noexcept-box"><span class="t-cc"><code>noexcept(noexcept(lhs.swap(rhs)))</code></span></div>
</div> <h3 id="Example">Ví dụ</h3> <div class="t-example"> <div class="cpp source-cpp"><pre data-language="cpp">Mẫu &lt;class ... loại&gt;</pre></div> <p>Nếu (sai)</p>
<div class="text source-text"><pre data-language="c">Cấu hình VOID SWAP (STD :: Variant &lt;type ...&gt; &amp; LHS,</pre></div> </div> <h3 id="Defect_reports">Báo cáo khiếm khuyết</h3> <p>Các báo cáo khiếm khuyết thay đổi hành vi sau đây được áp dụng hồi tố cho các tiêu chuẩn C ++ được công bố trước đó.</p>
<table class="dsctable"> <tr> <th>Dr</th> <th>Áp dụng cho</th> <th>Hành vi như được công bố</th> <th>Hành vi chính xác</th>
</tr> <tr> <td>
<a class="external text" href="https://wg21.link/P2231R1" rel="nofollow">P2231R1</a> </td> <td>C ++ 20</td> <td>
<code>swap</code>Bất kỳ ngoại lệ nào được ném bởi hàm tạo đã chọn của</td> <td>xây dựng giá trị chứa tại chỗ</td>
</tr>
</table> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="swap" title="cpp/utility/variant/swap"> <span class="t-lines"><span>Được xác định trong tiêu đề</span></span></a></div> </td> <td>xây dựng một giá trị trong biến thể, tại chỗ<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/variant/swap2">https://en.cppreference.com/w/cpp/utility/variant/swap2</a>
</p>
</div>
