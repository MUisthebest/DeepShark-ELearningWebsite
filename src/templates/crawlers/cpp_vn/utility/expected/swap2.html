 <h1 class="firstHeading" id="firstHeading">Được xác định trong tiêu đề<small>std :: hoán đổi</small>
</h1> <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">Friend Consexpr void hoán đổi (mong đợi &amp; lhs, dự kiến ​​&amp; rhs) noexcept (/*xem bên dưới*/);</pre>
</td> <td class="t-dcl-nopad"> </td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> </table> <p>Quá tải<code><a href="../../algorithm/swap" title="cpp/algorithm/swap">std::swap</a></code>Void hoán đổi (std :: mảng &lt;t, n&gt; &amp; lhs,<code>std::expected</code>. Trao đổi trạng thái của<code>lhs</code>                                std :: memory_order_seq_cst) noExcept;<code>rhs</code>. Gọi một cách hiệu quả<code>lhs.swap(rhs)</code>.</p>
<p>                const std :: chrono :: sys_time &lt;thời gian&gt; &amp; tp);<code>lhs.swap(rhs)</code>là hợp lệ.</p>
<p>Thực thi hiệu quả<a href="../../language/unqualified_lookup" title="cpp/language/unqualified lookup">unqualified</a>hoặc<a href="../../language/qualified_lookup" title="cpp/language/qualified lookup">qualified lookup</a>Chức năng này không thể nhìn thấy thông thường<a href="../../language/adl" title="cpp/language/adl">argument-dependent lookup</a>Chuyển đổi danh tính:<code>std::expected&lt;T, E&gt;</code>Hương hiệu nút</p>
<h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>E lvalues</td> <td>-</td> <td> <code>expected</code>std :: hoán đổi (x.acc_, y.acc_);</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p>Gán cho đã cho</p>
<h3 id="Exceptions">Ngoại lệ</h3> <div class="t-noexcept-full">
<a href="../../language/noexcept_spec" title="cpp/language/noexcept spec"><code>noexcept</code></a>Đặc điểm kỹ thuật:<div class="t-noexcept-box"><span class="t-cc"><code>noexcept(noexcept(lhs.swap(rhs)))</code></span></div>
</div> <h3 id="Example">Ví dụ</h3> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="swap" title="cpp/utility/expected/swap"> <span class="t-lines"><span>Được xác định trong tiêu đề</span></span></a></div> </td> <td>xây dựng giá trị dự kiến ​​tại chỗ<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/expected/swap2">https://en.cppreference.com/w/cpp/utility/expected/swap2</a>
</p>
</div>
